pgns/127489.js

```diff
@@ -136,3 +138,156 @@ module.exports = [
+  },
+  {
+    node: 'notifications.propulsion.%s.lowSystemVoltage',
+    message: '%s Low System Voltage'
+  },
+  {
+    node: 'notifications.propulsion.%s.lowCoolantLevel',
+    message: '%s Engine Low Coolant Level'
+  },
+  {
+    node: 'notifications.propulsion.%s.waterFlow',
+    message: '%s Engine Water Flow'
+  },
+  {
+    node: 'notifications.propulsion.%s.waterInFuel',
+    message: '%s Water in Fuel'
+  },
+  {
+    node: 'notifications.propulsion.%s.chargeIndicator',
+    message: '%s Engine Charge Indicator'
+  },
+  {
+    node: 'notifications.propulsion.%s.preheatIndicator',
+    message: '%s Preheat Indicator'
+  },
+  {
+    node: 'notifications.propulsion.%s.highBoostPressure',
+    message: '%s Engine High Boost Pressure'
+  },
+  {
+    node: 'notifications.propulsion.%s.revLimitExceeded',
+    message: '%s Engine Rev Limit Exceeded'
+  },
+  {
+    node: 'notifications.propulsion.%s.eGRSystem',
+    message: '%s Engine EGR System'
+  },
+  {
+    node: 'notifications.propulsion.%s.throttlePositionSensor',
+    message: '%s Engine Throttle Position Sensor'
+  },
+  {
+    node: 'notifications.propulsion.%s.emergencyStopMode',
+    message: '%s Engine Emergency Stop Mode'
+  }
+];
+
+var status2Notifications = [
+  { 
+    node: 'notifications.propulsion.%s.warningLevel1',
+    message: '%s Engine Warning Level 1'
+  },
+  {
+    node: 'notifications.propulsion.%s.warningLevel2',
+    message: '%s Engine Warning Level 2'
+  },
+  {
+    node: 'notifications.propulsion.%s.powerReduction',
+    message: '%s Engine Power Reduction'
+  },
+  {
+    node: 'notifications.propulsion.%s.maintenanceNeeded',
+    message: '%s Engine Maintenance Needed'
+  },
+  {
+    node: 'notifications.propulsion.%s.commError',
+    message: '%s Engine Comm Error'
+  },
+  {
+    node: 'notifications.propulsion.%s.subOrSecondaryThrottle',
+    message: '%s Engine Sub or Secondary Throttle'
+  },
+  {
+    node: 'notifications.propulsion.%s.neutralStartProtect',
+    message: '%s Neutral Start Protect'
+  },
+  {
+    node: 'notifications.propulsion.%s.shuttingDown',
+    message: '%s Engine Shutting Down'
+  }
+];
+
+
+function generateMappingsForStatus(field, notifications)
+{
+  notifications.forEach((notif,index) => {
+    var mapping = {
+      node: function(n2k) {
+        var engine = n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port' ? "port" : "starboard";
+        return util.format(notif.node, engine);
+      },
+      filter: function(n2k) { return typeof n2k.fields[field] !== 'undefined'; },
+      value: function(n2k, state) {
+        var engine = n2k.fields['Engine Instance'] === 'Single Engine or Dual Engine Port' ? "Port" : "Starboard";
+        
+        if ( n2k.fields[field] != 0 ) {
+          if ( n2k.fields[field] & (1<<index) ) {
+            if ( typeof state === 'undefined'
+                 || typeof state[notif.node] === 'undefined'
+                 || state[notif.node] != 'alarm') {
```

Comment:
What you describe about constant alarm deltas and silencing do not really belong in the n2k-signalk module: n2k-signalk should just convert n2k data to SK deltas. 

Whatever policies or silencing functionalities there are should take place somewhere downstream, in practice in the server. The server can receive input from the user and act on it and apply throttling policies.